soak-tests-common/src/main/java/com/hazelcast/jet/tests/common/AbstractSoakTest.java

```diff
@@ -83,19 +93,26 @@ protected final void run(String[] args) throws Exception {
         if (jet != null) {
             jet.shutdown();
         }
+        if (stableClusterClient != null) {
+            stableClusterClient.shutdown();
+        }
         if (instances != null) {
             Jet.shutdownAll();
         }
         logger.info("Finished OK");
         System.exit(0);
     }
 
-    protected abstract void init() throws Exception;
+    protected abstract void init(JetInstance client) throws Exception;
 
-    protected abstract void test() throws Throwable;
+    protected abstract void test(JetInstance client, String name) throws Throwable;
 
     protected abstract void teardown(Throwable t) throws Exception;
 
+    protected boolean runOnBothClusters() {
```

Comment:
Please add javadoc which will say that this method should be overridden if test should run on both clusters.